<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>netZooR application with TB dataset • netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="netZooR application with TB dataset">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/netZoo/netZooR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ApplicationwithTBdataset_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>netZooR application with TB dataset</h1>
                        <h4 class="author">Tian Wang</h4>
            
            <h4 class="date">2019-10-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/master/vignettes/ApplicationwithTBdataset.Rmd"><code>vignettes/ApplicationwithTBdataset.Rmd</code></a></small>
      <div class="hidden name"><code>ApplicationwithTBdataset.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>netZooR is an R package which consists of seven main algorithms and is able to construct, analyse and plot gene regulatory networks.</p>
<ul>
<li><p><strong>PANDA</strong>(Passing Attributes between Networks for Data Assimilation) is a message-passing model to gene regulatory network reconstruction. It integrates multiple sources of biological data, including protein-protein interaction, gene expression, and transcription factor binding motifs information to reconstruct genome-wide, condition-specific regulatory networks.<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064832">[Glass et al. 2013]</a></p></li>
<li><p><strong>LIONESS</strong>(Linear Interpolation to Obtain Network Estimates for Single Samples) is a method to estimate sample-specific regulatory networks by applying linear interpolation to the predictions made by existing aggregate network inference approaches.[[Kuijjer et al. 2019]]](<a href="https://www.sciencedirect.com/science/article/pii/S2589004219300872" class="uri">https://www.sciencedirect.com/science/article/pii/S2589004219300872</a>)</p></li>
<li><p><strong>CONDOR</strong> (COmplex Network Description Of Regulators) implements methods to cluster biapartite networks and estimatiing the contribution of each node to its community’s modularity.<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005033">[Platig et al. 2016]</a></p></li>
<li><p><strong>ALPACA</strong>(ALtered Partitions Across Community Architectures) is a method to compare two genome-scale networks derived from different phenotypic states to identify condition-specific modules. <a href="https://www.nature.com/articles/s41540-018-0052-5">[Padi and Quackenbush 2018]</a></p></li>
<li><p><strong>SAMBAR</strong>(Subtyping Agglomerated Mutations By Annotation Relations) is a method to identify subtypes based on somatic mutation data.<a href="https://www.nature.com/articles/s41416-018-0109-7">[Kuijjer et al.]</a>.</p></li>
<li><p><strong>MONSTER</strong>(Modeling Network State Transitions from Expression and Regulatory data)<a href="https://doi.org/10.1186/s12918-017-0517-y">[Schlauch et al.]</a>: infers transcription factor which drivers of cell state conditions at the gene regulatory network level.</p></li>
<li><p><strong>OTTER</strong>(Optimization to Estimate Regulation) [publication in preparation]: models gene regulation estimation as a graph matrching problem</p></li>
</ul>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div id="prerequisites" class="section level3">
<h3 class="hasAnchor">
<a href="#prerequisites" class="anchor"></a>Prerequisites</h3>
<p>Using this pacakage requires <a href="https://www.python.org/downloads/"><strong>Python</strong></a> (3.X) and some <a href="#required-python-libraries">Python libraries</a>, <a href="https://cran.r-project.org/"><strong>R</strong></a> (&gt;= 3.3.3), and stable <strong>Internet access</strong>.</p>
<p>Some plotting functions will require the <a href="https://cytoscape.org/"><strong>Cytoscape</strong></a> installed.</p>
</div>
<div id="required-python-libraries" class="section level3">
<h3 class="hasAnchor">
<a href="#required-python-libraries" class="anchor"></a>Required Python libraries</h3>
<p>How to install Python libraries depends varies from different platforms. More instructions could be find <a href="https://packaging.python.org/tutorials/installing-packages/">here</a>.</p>
<p>The following Python libraries (or packages) are required by running PANDA and LIONESS algorithms:</p>
<p>The required Python packages are: <a href="https://pandas.pydata.org/">pandas</a>, <a href="http://www.numpy.org/">numpy</a>, <a href="https://networkx.github.io/">networkx</a>, <a href="https://matplotlib.org/api/pyplot_api.html">matplotlib.pyplot</a>.</p>
</div>
<div id="installing" class="section level3">
<h3 class="hasAnchor">
<a href="#installing" class="anchor"></a>Installing</h3>
<p>This package could be downloaded via <code><a href="https://rdrr.io/pkg/devtools/man/remote-reexports.html">install_github()</a></code> function from <code>devtools</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools") </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="co"># install netZooR pkg with vignettes, otherwise remove the "build_vignettes = TRUE" argument.</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/devtools/man/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"netZoo/netZooR"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridisLite/">viridisLite</a></span><span class="op">)</span><span class="co">#To visualize communities</span></code></pre></div>
</div>
</div>
<div id="data-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#data-resources" class="anchor"></a>Data Resources</h2>
<div id="motif-data" class="section level3">
<h3 class="hasAnchor">
<a href="#motif-data" class="anchor"></a>Motif data</h3>
<p>Here is some pre-prepared specie-sepcific PANDA-ready transcription factor binding motifs data stored in our AWS bucket <a href="https://s3.console.aws.amazon.com/s3/buckets/netzoo/netZooR/example_datasets/PANDA_ready_motif_prior/?region=us-east-2&amp;tab=overview" class="uri">https://s3.console.aws.amazon.com/s3/buckets/netzoo/netZooR/example_datasets/PANDA_ready_motif_prior/?region=us-east-2&amp;tab=overview</a>, which are derived from motif scan and motif info files located on <a href="https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resourcesby" class="uri">https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resourcesby</a> .</p>
</div>
<div id="ppi" class="section level3">
<h3 class="hasAnchor">
<a href="#ppi" class="anchor"></a>PPI</h3>
<p>This package includes a function <code>source.PPI</code> may source a Protein-Protein Interactions (PPI) througt STRING database given a list of proteins of interest. The <a href="http://www.bioconductor.org/packages/release/bioc/html/STRINGdb.html">STRINGdb</a> is already loaded while loading netZooR.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># TF is a data frame with single column filled with TFs of Mycobacterium tuberculosis H37Rv.</span>
<span class="va">PPI</span> <span class="op">&lt;-</span> <span class="fu">source.PPI</span><span class="op">(</span><span class="va">TF</span>, STRING.version<span class="op">=</span><span class="st">"10"</span>, species.index<span class="op">=</span><span class="fl">83332</span>, score_threshold<span class="op">=</span><span class="fl">0</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="running-the-sample-tb-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#running-the-sample-tb-datasets" class="anchor"></a>Running the sample TB datasets</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/netZoo/netZooR">netZooR</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: igraph</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'igraph'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     decompose, spectrum</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     union</span>
<span class="co">#&gt; Loading required package: reticulate</span>
<span class="co">#&gt; Loading required package: pandaR</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:igraph':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     normalize, path, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; Loading required package: yarn</span>
<span class="co">#&gt; Setting options('download.file.method.GEOquery'='auto')</span>
<span class="co">#&gt; Setting options('GEOquery.inmemory.gpl'=FALSE)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span></code></pre></div>
<p>Accessing the help pages for the usage of core functions.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">pandaPy</span>
<span class="op">?</span><span class="va">createCondorObject</span>
<span class="op">?</span><span class="va">pandaToCondorObject</span>
<span class="op">?</span><span class="va">lionessPy</span>
<span class="op">?</span><span class="va">alpaca</span>
<span class="op">?</span><span class="va">pandaToAlpaca</span>
<span class="op">?</span><span class="va">sambar</span></code></pre></div>
<p>This package will invoke the Python in R environment through reticulate package. Configure which version of Python to use if necessary, here in netZooR, Python 3.X is required. More details can be found <a href="https://cran.r-project.org/web/packages/reticulate/vignettes/versions.html">here</a></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#check your Python configuration and the specific version of Python in use currently</span>
<span class="fu">py_config</span><span class="op">(</span><span class="op">)</span>

<span class="co"># reset to Python 3.X if necessary, like below:</span>
<span class="fu">use_python</span><span class="op">(</span><span class="st">"/usr/local/bin/python3"</span><span class="op">)</span></code></pre></div>
<p>The previous command is necessary to bind R to Python since we are calling PANDA from Python because netZooPy has an optimized implementation of PANDA. Check <a href="http://netbooks.networkmedicine.org/user/marouenbg/notebooks/netZooR/panda_gtex_tutorial_server.ipynb">this tutorial</a> for an example using a pure R implementation of PANDA. Use example data sets within package to test this package. Refer to four input datasets files: one TB expression dataset control group , one TB expression dataset treated, one transcription factor binding motifs dataset, and one protein-protein interaction datasets from either inst/extdat or AWS.</p>
<p>retrieve the file path of these files came with the netZooR package.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># retrieve the file path of these files</span>
<span class="va">treated_expression_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expr4_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">control_expression_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expr10_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">motif_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"chip_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">ppi_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ppi_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="panda-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#panda-algorithm" class="anchor"></a>PANDA algorithm</h3>
<p>Assign the above file paths to flag <code>e</code>(refers to “expression dataset”), <code>m</code>(refers to “motif dataset”), and <code>ppi</code>(refers to “PPI” dataset), respectively. Then set option <code>rm_missing</code> to <code>TRUE</code> to run <strong>PANDA</strong> to generate an aggregate network without unmatched TF and genes.</p>
<p>Repeat with control group.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">treated_all_panda_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaPy.html">pandaPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="va">treated_expression_file_path</span>, motif_file <span class="op">=</span> <span class="va">motif_file_path</span>, ppi_file<span class="op">=</span> <span class="va">ppi_file_path</span>,modeProcess<span class="op">=</span><span class="st">"legacy"</span>,  remove_missing <span class="op">=</span> <span class="cn">TRUE</span> <span class="op">)</span>
<span class="va">control_all_panda_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaPy.html">pandaPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="va">control_expression_file_path</span>,motif_file <span class="op">=</span> <span class="va">motif_file_path</span>, ppi_file<span class="op">=</span> <span class="va">ppi_file_path</span>,modeProcess<span class="op">=</span><span class="st">"legacy"</span>,  remove_missing <span class="op">=</span> <span class="cn">TRUE</span> <span class="op">)</span></code></pre></div>
<p>Vector <code>treated_all_panda_result</code> and vector <code>control_all_panda_result</code> below are large lists with three elements: the entire PANDA network, indegree (“to” nodes) nodes and score, outdegree (“from” nodes) nodes and score. Use <code>$panda</code>,<code>$indegree</code> and <code>$outdegree</code> to access each list item resepctively.</p>
<p>Use <code>$panda</code>to access the entire PANDA network.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">treated_net</span> <span class="op">&lt;-</span> <span class="va">treated_all_panda_result</span><span class="op">$</span><span class="va">panda</span>
<span class="va">control_net</span> <span class="op">&lt;-</span> <span class="va">control_all_panda_result</span><span class="op">$</span><span class="va">panda</span></code></pre></div>
</div>
<div id="panda-cytoscape-plotting" class="section level3">
<h3 class="hasAnchor">
<a href="#panda-cytoscape-plotting" class="anchor"></a>PANDA Cytoscape Plotting</h3>
<p>Cytoscape is an interactivity network visualization tool highly recommanded to explore the PANDA network. Before using this function <code>plot.panda.in.cytoscape</code>, please install and launch Cytoscape (3.6.1 or greater) and keep it running whenever using.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># select top 1000 edges in PANDA network by edge weight.</span>
<span class="va">panda.net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">treated_net</span><span class="op">[</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">order</a></span><span class="op">(</span><span class="va">control_net</span><span class="op">$</span><span class="va">force</span>,decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>, <span class="fl">1000</span><span class="op">)</span>
 
<span class="co"># run this function to create a network in Cytoscape.</span>
<span class="fu">vis.panda.in.cytoscape</span><span class="op">(</span><span class="va">panda.net</span>, network.name<span class="op">=</span><span class="st">"PANDA"</span><span class="op">)</span></code></pre></div>
</div>
<div id="lioness-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#lioness-algorithm" class="anchor"></a>LIONESS Algorithm</h3>
<p>How to run LIONESS is mostly idential with method how to run PANDA in this package, unless the return values of <code><a href="../reference/lionessPy.html">lionessPy()</a></code> is a data frame where first two columns represent TFs (regulators) and Genes (targets) while the rest columns represent each sample. each cell filled with estimated score calculated by LIONESS.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run LIONESS algorithm for the first two samples</span>
<span class="co"># removing start_sample and end_sample arguments to generate whole LIONESS network with all samples.</span>
<span class="va">control_lioness_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lionessPy.html">lionessPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="va">control_expression_file_path</span>,motif_file <span class="op">=</span> <span class="va">motif_file_path</span>, ppi_file<span class="op">=</span> <span class="va">ppi_file_path</span>,modeProcess<span class="op">=</span><span class="st">"legacy"</span>,  remove_missing <span class="op">=</span> <span class="cn">TRUE</span>, start_sample<span class="op">=</span><span class="fl">1</span>, end_sample<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
<div id="condor-algorithm-and-plotting" class="section level3">
<h3 class="hasAnchor">
<a href="#condor-algorithm-and-plotting" class="anchor"></a>CONDOR Algorithm and plotting</h3>
<p>PANDA network can simply be converted into condor.object by <code><a href="../reference/pandaToCondorObject.html">pandaToCondorObject(panda.net, threshold)</a></code> Defaults option <code>threshold</code> is the average of [median weight of non-prior edges] and [median weight of prior edges], all weights mentioned previous are transformationed with formula <code>w'=ln(e^w+1)</code> before calculating the median and average. But all the edges selected will remain the orginal weights calculated by PANDA.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">treated_condor_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaToCondorObject.html">pandaToCondorObject</a></span><span class="op">(</span><span class="va">treated_net</span>, threshold <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p>The communities structure can be plotted by igraph.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridisLite/">viridisLite</a></span><span class="op">)</span>
<span class="va">treated_condor_object</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/condorCluster.html">condorCluster</a></span><span class="op">(</span><span class="va">treated_condor_object</span>,project <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; [1] "modularity of projected graph 0.306134881290028"</span>
<span class="co">#&gt; [1] "Q = 0.306345744728624"</span>
<span class="co">#&gt; [1] "Q = 0.306345744728624"</span>
<span class="va">treated_color_num</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">treated_condor_object</span><span class="op">$</span><span class="va">red.memb</span><span class="op">$</span><span class="va">com</span><span class="op">)</span>
<span class="va">treated_color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html">viridis</a></span><span class="op">(</span><span class="va">treated_color_num</span>, alpha <span class="op">=</span> <span class="fl">1</span>, begin <span class="op">=</span> <span class="fl">0</span>, end <span class="op">=</span> <span class="fl">1</span>, direction <span class="op">=</span> <span class="fl">1</span>, option <span class="op">=</span> <span class="st">"D"</span><span class="op">)</span>
<span class="fu"><a href="../reference/condorPlotCommunities.html">condorPlotCommunities</a></span><span class="op">(</span><span class="va">treated_condor_object</span>, color_list<span class="op">=</span><span class="va">treated_color</span>, point.size<span class="op">=</span><span class="fl">0.04</span>, xlab<span class="op">=</span><span class="st">"Genes"</span>, ylab<span class="op">=</span><span class="st">"TFs"</span><span class="op">)</span></code></pre></div>
<p><img src="ApplicationwithTBdataset_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div id="alpaca-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#alpaca-algorithm" class="anchor"></a>ALPACA Algorithm</h3>
<p>ALPACA community structure can also be generated from two PANDA network by <code>pandaToAlpaca</code></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alpaca</span><span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaToAlpaca.html">pandaToAlpaca</a></span><span class="op">(</span><span class="va">treated_net</span>, <span class="va">control_net</span>, <span class="cn">NULL</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; [1] "Detecting communities in control network..."</span>
<span class="co">#&gt; Weights detected. Building condor object with weighted edges.</span>
<span class="co">#&gt; [1] "modularity of projected graph 0.482467704493567"</span>
<span class="co">#&gt; [1] "Q = 0.49405163585898"</span>
<span class="co">#&gt; [1] "Q = 0.49405163585898"</span>
<span class="co">#&gt; [1] "Computing differential modularity matrix..."</span>
<span class="co">#&gt; [1] "Computing differential modules..."</span>
<span class="co">#&gt; [1] "Merging 2644 communities"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; [1] "Merging 139 communities"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; [1] "Merging 18 communities"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "Merging 14 communities"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] "Computing node scores..."</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; [1] 6</span>
<span class="co">#&gt; [1] 7</span>
<span class="co">#&gt; [1] 8</span>
<span class="co">#&gt; [1] 9</span>
<span class="co">#&gt; [1] 10</span>
<span class="co">#&gt; [1] 11</span>
<span class="co">#&gt; [1] 12</span>
<span class="co">#&gt; [1] 13</span>
<span class="co">#&gt; [1] 14</span></code></pre></div>
</div>
</div>
<div id="more-tutorials" class="section level2">
<h2 class="hasAnchor">
<a href="#more-tutorials" class="anchor"></a>More tutorials</h2>
<p>Browse with <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes("netZooR")</a></code></p>
</div>
<div id="information" class="section level2">
<h2 class="hasAnchor">
<a href="#information" class="anchor"></a>Information</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.1 (2021-08-10)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Mojave 10.14.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] viridisLite_0.4.0   netZooR_1.1.16      yarn_1.18.0        </span>
<span class="co">#&gt; [4] pandaR_1.19.5       Biobase_2.52.0      BiocGenerics_0.38.0</span>
<span class="co">#&gt; [7] reticulate_1.22     igraph_1.2.6       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] utf8_1.2.2                  proto_1.0.0                </span>
<span class="co">#&gt;   [3] R.utils_2.11.0              RUnit_0.4.32               </span>
<span class="co">#&gt;   [5] tidyselect_1.1.1            RSQLite_2.2.8              </span>
<span class="co">#&gt;   [7] AnnotationDbi_1.54.1        grid_4.1.1                 </span>
<span class="co">#&gt;   [9] BiocParallel_1.26.2         base64url_1.4              </span>
<span class="co">#&gt;  [11] munsell_0.5.0               codetools_0.2-18           </span>
<span class="co">#&gt;  [13] ragg_1.1.3                  preprocessCore_1.54.0      </span>
<span class="co">#&gt;  [15] chron_2.3-56                pbdZMQ_0.3-5               </span>
<span class="co">#&gt;  [17] colorspace_2.0-2            Category_2.58.0            </span>
<span class="co">#&gt;  [19] filelock_1.0.2              highr_0.9                  </span>
<span class="co">#&gt;  [21] dplR_1.7.2                  knitr_1.36                 </span>
<span class="co">#&gt;  [23] uuid_0.1-4                  stats4_4.1.1               </span>
<span class="co">#&gt;  [25] MatrixGenerics_1.4.3        repr_1.1.3                 </span>
<span class="co">#&gt;  [27] GenomeInfoDbData_1.2.6      bit64_4.0.5                </span>
<span class="co">#&gt;  [29] rhdf5_2.36.0                rprojroot_2.0.2            </span>
<span class="co">#&gt;  [31] downloader_0.4              vctrs_0.3.8                </span>
<span class="co">#&gt;  [33] generics_0.1.0              xfun_0.26                  </span>
<span class="co">#&gt;  [35] BiocFileCache_2.0.0         R6_2.5.1                   </span>
<span class="co">#&gt;  [37] doParallel_1.0.16           GenomeInfoDb_1.28.4        </span>
<span class="co">#&gt;  [39] illuminaio_0.34.0           RJSONIO_1.3-1.6            </span>
<span class="co">#&gt;  [41] locfit_1.5-9.4              bitops_1.0-7               </span>
<span class="co">#&gt;  [43] rhdf5filters_1.4.0          cachem_1.0.6               </span>
<span class="co">#&gt;  [45] reshape_0.8.8               DelayedArray_0.18.0        </span>
<span class="co">#&gt;  [47] assertthat_0.2.1            BiocIO_1.2.0               </span>
<span class="co">#&gt;  [49] scales_1.1.1                nnet_7.3-16                </span>
<span class="co">#&gt;  [51] gtable_0.3.0                rlang_0.4.11               </span>
<span class="co">#&gt;  [53] genefilter_1.74.0           systemfonts_1.0.2          </span>
<span class="co">#&gt;  [55] splines_4.1.1               rtracklayer_1.52.1         </span>
<span class="co">#&gt;  [57] GEOquery_2.60.0             hexbin_1.28.2              </span>
<span class="co">#&gt;  [59] reshape2_1.4.4              yaml_2.2.1                 </span>
<span class="co">#&gt;  [61] backports_1.2.1             GenomicFeatures_1.44.2     </span>
<span class="co">#&gt;  [63] RBGL_1.68.0                 tools_4.1.1                </span>
<span class="co">#&gt;  [65] nor1mix_1.3-0               ggplot2_3.3.5              </span>
<span class="co">#&gt;  [67] ellipsis_0.3.2              gplots_3.1.1               </span>
<span class="co">#&gt;  [69] jquerylib_0.1.4             RColorBrewer_1.1-2         </span>
<span class="co">#&gt;  [71] ggdendro_0.1.22             siggenes_1.66.0            </span>
<span class="co">#&gt;  [73] STRINGdb_2.4.1              gsubfn_0.7                 </span>
<span class="co">#&gt;  [75] hash_2.2.6.1                Rcpp_1.0.7                 </span>
<span class="co">#&gt;  [77] plyr_1.8.6                  base64enc_0.1-3            </span>
<span class="co">#&gt;  [79] sparseMatrixStats_1.4.2     progress_1.2.2             </span>
<span class="co">#&gt;  [81] zlibbioc_1.38.0             purrr_0.3.4                </span>
<span class="co">#&gt;  [83] RCurl_1.98-1.5              prettyunits_1.1.1          </span>
<span class="co">#&gt;  [85] sqldf_0.4-11                openssl_1.4.5              </span>
<span class="co">#&gt;  [87] bumphunter_1.34.0           S4Vectors_0.30.2           </span>
<span class="co">#&gt;  [89] cluster_2.1.2               SummarizedExperiment_1.22.0</span>
<span class="co">#&gt;  [91] here_1.0.1                  fs_1.5.0                   </span>
<span class="co">#&gt;  [93] magrittr_2.0.1              data.table_1.14.2          </span>
<span class="co">#&gt;  [95] matrixStats_0.61.0          hms_1.1.1                  </span>
<span class="co">#&gt;  [97] evaluate_0.14               xtable_1.8-4               </span>
<span class="co">#&gt;  [99] XML_3.99-0.8                mclust_5.4.7               </span>
<span class="co">#&gt; [101] IRanges_2.26.0              compiler_4.1.1             </span>
<span class="co">#&gt; [103] biomaRt_2.48.3              minfi_1.38.0               </span>
<span class="co">#&gt; [105] tibble_3.1.5                RCy3_2.12.4                </span>
<span class="co">#&gt; [107] KernSmooth_2.23-20          crayon_1.4.1               </span>
<span class="co">#&gt; [109] R.oo_1.24.0                 htmltools_0.5.2            </span>
<span class="co">#&gt; [111] GOstats_2.58.0              mgcv_1.8-36                </span>
<span class="co">#&gt; [113] tzdb_0.1.2                  tidyr_1.1.4                </span>
<span class="co">#&gt; [115] DBI_1.1.1                   dbplyr_2.1.1               </span>
<span class="co">#&gt; [117] MASS_7.3-54                 rappdirs_0.3.3             </span>
<span class="co">#&gt; [119] Matrix_1.3-4                readr_2.0.2                </span>
<span class="co">#&gt; [121] permute_0.9-5               quantro_1.26.0             </span>
<span class="co">#&gt; [123] uchardet_1.1.0              R.methodsS3_1.8.1          </span>
<span class="co">#&gt; [125] quadprog_1.5-8              GenomicRanges_1.44.0       </span>
<span class="co">#&gt; [127] pkgconfig_2.0.3             pkgdown_1.6.1              </span>
<span class="co">#&gt; [129] GenomicAlignments_1.28.0    signal_0.7-7               </span>
<span class="co">#&gt; [131] IRdisplay_1.0               xml2_1.3.2                 </span>
<span class="co">#&gt; [133] foreach_1.5.1               annotate_1.70.0            </span>
<span class="co">#&gt; [135] penalized_0.9-51            bslib_0.3.1                </span>
<span class="co">#&gt; [137] rngtools_1.5                multtest_2.48.0            </span>
<span class="co">#&gt; [139] beanplot_1.2                XVector_0.32.0             </span>
<span class="co">#&gt; [141] AnnotationForge_1.34.0      doRNG_1.8.2                </span>
<span class="co">#&gt; [143] scrime_1.3.5                stringr_1.4.0              </span>
<span class="co">#&gt; [145] digest_0.6.28               vegan_2.5-7                </span>
<span class="co">#&gt; [147] graph_1.70.0                Biostrings_2.60.2          </span>
<span class="co">#&gt; [149] rmarkdown_2.11              base64_2.0                 </span>
<span class="co">#&gt; [151] edgeR_3.34.1                DelayedMatrixStats_1.14.3  </span>
<span class="co">#&gt; [153] GSEABase_1.54.0             restfulr_0.0.13            </span>
<span class="co">#&gt; [155] curl_4.3.2                  Rsamtools_2.8.0            </span>
<span class="co">#&gt; [157] gtools_3.9.2                rjson_0.2.20               </span>
<span class="co">#&gt; [159] lifecycle_1.0.1             nlme_3.1-152               </span>
<span class="co">#&gt; [161] jsonlite_1.7.2              Rhdf5lib_1.14.2            </span>
<span class="co">#&gt; [163] desc_1.4.0                  askpass_1.1                </span>
<span class="co">#&gt; [165] limma_3.48.3                fansi_0.5.0                </span>
<span class="co">#&gt; [167] pillar_1.6.3                lattice_0.20-44            </span>
<span class="co">#&gt; [169] plotrix_3.8-2               KEGGREST_1.32.0            </span>
<span class="co">#&gt; [171] fastmap_1.1.0               httr_1.4.2                 </span>
<span class="co">#&gt; [173] survival_3.2-11             GO.db_3.13.0               </span>
<span class="co">#&gt; [175] glue_1.4.2                  png_0.1-7                  </span>
<span class="co">#&gt; [177] iterators_1.0.13            bit_4.0.4                  </span>
<span class="co">#&gt; [179] Rgraphviz_2.36.0            stringi_1.7.5              </span>
<span class="co">#&gt; [181] sass_0.4.0                  HDF5Array_1.20.0           </span>
<span class="co">#&gt; [183] blob_1.2.2                  textshaping_0.3.5          </span>
<span class="co">#&gt; [185] org.Hs.eg.db_3.13.0         caTools_1.18.2             </span>
<span class="co">#&gt; [187] memoise_2.0.0               IRkernel_1.2               </span>
<span class="co">#&gt; [189] dplyr_1.0.7</span></code></pre></div>
</div>
<div id="note" class="section level2">
<h2 class="hasAnchor">
<a href="#note" class="anchor"></a>Note</h2>
<p>If there is an error like <code>Error in fetch(key) : lazy-load database.rdb' is corrupt</code> when accessing the help pages of functions in this package after being loaded. It’s <a href="https://github.com/r-lib/devtools/issues/1660">a limitation of base R</a> and has not been solved yet. Restart R session and re-load this package will help.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
