<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER) — monster • netZooR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="MOdeling Network State Transitions from Expression and Regulatory data (MONSTER) — monster" />
<meta property="og:description" content="This function runs the MONSTER algorithm.  Biological states are characterized by distinct patterns 
of gene expression that reflect each phenotype's active cellular processes. 
Driving these phenotypes are gene regulatory networks in which transcriptions factors control 
when and to what degree individual genes are expressed. Phenotypic transitions, such as those that 
occur when disease arises from healthy tissue, are associated with changes in these  networks. 
MONSTER is an approach to understanding these transitions. MONSTER models phenotypic-specific 
regulatory networks and then estimates a &quot;transition matrix&quot; that converts one state to another. 
By examining the properties of the transition matrix, we can gain insight into regulatory 
changes associated with phenotypic state transition.
Important note: the direct regulatory network observed from gene expression is currently
implemented as a regular correlation as opposed to the partial correlation described 
in the paper.
Citation: Schlauch, Daniel, et al. &quot;Estimating drivers of cell state transitions using gene regulatory network models.&quot; 
BMC systems biology 11.1 (2017): 139. https://doi.org/10.1186/s12918-017-0517-y" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/netZoo/netZooR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</h1>
    <small class="dont-index">Source: <a href='https://github.com/netZoo/netZooR/blob/master/R/MONSTER.R'><code>R/MONSTER.R</code></a></small>
    <div class="hidden name"><code>monster.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function runs the MONSTER algorithm.  Biological states are characterized by distinct patterns 
of gene expression that reflect each phenotype's active cellular processes. 
Driving these phenotypes are gene regulatory networks in which transcriptions factors control 
when and to what degree individual genes are expressed. Phenotypic transitions, such as those that 
occur when disease arises from healthy tissue, are associated with changes in these  networks. 
MONSTER is an approach to understanding these transitions. MONSTER models phenotypic-specific 
regulatory networks and then estimates a "transition matrix" that converts one state to another. 
By examining the properties of the transition matrix, we can gain insight into regulatory 
changes associated with phenotypic state transition.
Important note: the direct regulatory network observed from gene expression is currently
implemented as a regular correlation as opposed to the partial correlation described 
in the paper.
Citation: Schlauch, Daniel, et al. "Estimating drivers of cell state transitions using gene regulatory network models." 
BMC systems biology 11.1 (2017): 139. https://doi.org/10.1186/s12918-017-0517-y</p>
    </div>

    <pre class="usage"><span class='fu'>monster</span><span class='op'>(</span>
  <span class='va'>expr</span>,
  <span class='va'>design</span>,
  <span class='va'>motif</span>,
  nullPerms <span class='op'>=</span> <span class='fl'>100</span>,
  ni_method <span class='op'>=</span> <span class='st'>"BERE"</span>,
  ni.coefficient.cutoff <span class='op'>=</span> <span class='cn'>NA</span>,
  numMaxCores <span class='op'>=</span> <span class='fl'>1</span>,
  outputDir <span class='op'>=</span> <span class='cn'>NA</span>,
  alphaw <span class='op'>=</span> <span class='fl'>0.5</span>,
  mode <span class='op'>=</span> <span class='st'>"buildNet"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>expr</th>
      <td><p>Gene Expression dataset, can be matrix or data.frame of expression values or ExpressionSet.</p></td>
    </tr>
    <tr>
      <th>design</th>
      <td><p>Binary vector indicating case control partition. 1 for case and 0 for control.</p></td>
    </tr>
    <tr>
      <th>motif</th>
      <td><p>Regulatory data.frame consisting of three columns.  For each row, a transcription factor (column 1) 
regulates a gene (column 2) with a defined strength (column 3), usually taken to be 0 or 1</p></td>
    </tr>
    <tr>
      <th>nullPerms</th>
      <td><p>number of random permutations to run (default 100).  Set to 0 to only 
calculate observed transition matrix. When mode is is 'buildNet' it randomly permutes the case and control expression
samples, if mode is 'regNet' it will randomly permute the case and control networks.</p></td>
    </tr>
    <tr>
      <th>ni_method</th>
      <td><p>String to indicate algorithm method.  Must be one of "bere","pearson","cd","lda", or "wcd". Default is "bere"</p></td>
    </tr>
    <tr>
      <th>ni.coefficient.cutoff</th>
      <td><p>numeric to specify a p-value cutoff at the network
inference step.  Default is NA, indicating inclusion of all coefficients.</p></td>
    </tr>
    <tr>
      <th>numMaxCores</th>
      <td><p>requires doParallel, foreach.  Runs MONSTER in parallel computing 
environment.  Set to 1 to avoid parallelization, NA will take the default parallel pool in the computer.</p></td>
    </tr>
    <tr>
      <th>outputDir</th>
      <td><p>character vector specifying a directory or path in which 
which to save MONSTER results, default is NA and results are not saved.</p></td>
    </tr>
    <tr>
      <th>alphaw</th>
      <td><p>A weight parameter between 0 and 1 specifying proportion of weight 
to give to indirect compared to direct evidence. The default is 0.5 to give an 
equal weight to direct and indirect evidence.</p></td>
    </tr>
    <tr>
      <th>mode</th>
      <td><p>A parameter telling whether to build the regulatory networks ('buildNet') or to use provided regulatory networks
('regNet'). If set to 'regNet', then the parameters motif, ni_method, ni.coefficient.cutoff, and alphaw will be set to NA.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class "monsterAnalysis" containing results</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Example with the network reconstruction step</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>yeast</span><span class='op'>)</span>
<span class='va'>design</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>20</span><span class='op'>)</span>,<span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='cn'>NA</span>,<span class='fl'>10</span><span class='op'>)</span>,<span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>20</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>yeast</span><span class='op'>$</span><span class='va'>exp.cc</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>yeast</span><span class='op'>$</span><span class='va'>exp.cc</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/mean.html'>mean</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='va'>yeast</span><span class='op'>$</span><span class='va'>exp.cc</span><span class='op'>)</span>,na.rm<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>monsterRes</span> <span class='op'>&lt;-</span> <span class='fu'>monster</span><span class='op'>(</span><span class='va'>yeast</span><span class='op'>$</span><span class='va'>exp.cc</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>500</span>,<span class='op'>]</span>, <span class='va'>design</span>, <span class='va'>yeast</span><span class='op'>$</span><span class='va'>motif</span>, nullPerms<span class='op'>=</span><span class='fl'>10</span>, numMaxCores<span class='op'>=</span><span class='fl'>1</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "11 network transitions to be estimated"
#&gt; [1] "Running iteration 1"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 0.0002229214 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 4.410744e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 1"
#&gt; [1] "Running iteration 2"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 0.0002498627 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 6.198883e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 2"
#&gt; [1] "Running iteration 3"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 5.602837e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 8.797646e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 3"
#&gt; [1] "Running iteration 4"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 6.389618e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 9.799004e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 4"
#&gt; [1] "Running iteration 5"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 7.104874e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 7.200241e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 5"
#&gt; [1] "Running iteration 6"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 4.816055e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 7.486343e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 6"
#&gt; [1] "Running iteration 7"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 9.894371e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; ................[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 4.315376e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 7"
#&gt; [1] "Running iteration 8"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 3.98159e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 4.506111e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 8"
#&gt; [1] "Running iteration 9"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 4.410744e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 4.410744e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 9"
#&gt; [1] "Running iteration 10"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 4.887581e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 5.412102e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Using OLS method"
#&gt; [1] "Finished running iteration 10"
#&gt; [1] "Running iteration 11"
#&gt; [1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 0.0001108646 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; .....</div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</span></div><div class='output co'>#&gt; ...........[1] "Initializing and validating"
#&gt; [1] "Verified adequate samples, calculating correlation matrix"
#&gt; Time difference of 5.507469e-05 secs
#&gt; [1] "More data cleaning"
#&gt; [1] "Main calculation"
#&gt; ................[1] "Using OLS method"
#&gt; [1] "Finished running iteration 11"
#&gt; Time difference of 4.559445 secs</div><div class='input'><span class='co'># Example with provided networks</span>
<span class='co'># \donttest{</span>
<span class='va'>pandaResult</span> <span class='op'>&lt;-</span> <span class='fu'>panda</span><span class='op'>(</span><span class='va'>pandaToyData</span><span class='op'>$</span><span class='va'>motif</span>, <span class='va'>pandaToyData</span><span class='op'>$</span><span class='va'>expression</span>, <span class='va'>pandaToyData</span><span class='op'>$</span><span class='va'>ppi</span><span class='op'>)</span>
<span class='va'>case</span><span class='op'>=</span><span class='fu'>getRegNet</span><span class='op'>(</span><span class='va'>pandaResult</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in getRegNet(pandaResult): could not find function "getRegNet"</span></div><div class='input'><span class='va'>nelemReg</span><span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='fu'>getRegNet</span><span class='op'>(</span><span class='va'>pandaResult</span><span class='op'>)</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>*</span><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='fu'>getRegNet</span><span class='op'>(</span><span class='va'>pandaResult</span><span class='op'>)</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span>
</div><div class='output co'>#&gt; <span class='error'>Error in getRegNet(pandaResult): could not find function "getRegNet"</span></div><div class='input'><span class='va'>nGenes</span><span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>getRegNet</span><span class='op'>(</span><span class='va'>pandaResult</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'colnames': could not find function "getRegNet"</span></div><div class='input'><span class='va'>control</span><span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/Exponential.html'>rexp</a></span><span class='op'>(</span><span class='va'>nelemReg</span>, rate<span class='op'>=</span><span class='fl'>.1</span><span class='op'>)</span>, ncol<span class='op'>=</span><span class='va'>nGenes</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in rexp(nelemReg, rate = 0.1): object 'nelemReg' not found</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>control</span><span class='op'>)</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>case</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'colnames': object 'case' not found</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>control</span><span class='op'>)</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>case</span><span class='op'>)</span> 
</div><div class='output co'>#&gt; <span class='error'>Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'rownames': object 'case' not found</span></div><div class='input'><span class='va'>expr</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/cbind.html'>cbind</a></span><span class='op'>(</span><span class='va'>control</span>,<span class='va'>case</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': object 'control' not found</span></div><div class='input'><span class='va'>design</span><span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='va'>nGenes</span><span class='op'>)</span>,<span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='va'>nGenes</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in eval(expr, envir, enclos): object 'nGenes' not found</span></div><div class='input'><span class='va'>monsterRes</span> <span class='op'>&lt;-</span> <span class='fu'>monster</span><span class='op'>(</span><span class='va'>expr</span>, <span class='va'>design</span>, motif<span class='op'>=</span><span class='cn'>NA</span>, nullPerms<span class='op'>=</span><span class='fl'>10</span>, numMaxCores<span class='op'>=</span><span class='fl'>1</span>, mode<span class='op'>=</span><span class='st'>'regNet'</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in is.data.frame(expr): object 'expr' not found</span></div><div class='input'><span class='co'># }</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


