<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Identify differential edges in two PANDA networks — pandaDiffEdges • netZooR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Identify differential edges in two PANDA networks — pandaDiffEdges" />
<meta property="og:description" content="To determine the probability that an edge is &quot;different&quot; between the networks, 
we first subtracted the z-score weight values estimated by PANDA for the two networks and then determined the value of the inverse cumulative distribution for this difference. 
The product of these two probabilities represents the probability than an edge is both &quot;supported&quot; and &quot;different.&quot;
We select edges for which this combined probability is greater than a threshold probability (default value is 0.8)." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/netZoo/netZooR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Identify differential edges in two PANDA networks</h1>
    <small class="dont-index">Source: <a href='https://github.com/netZoo/netZooR/blob/master/R/pandaDiffEdges.R'><code>R/pandaDiffEdges.R</code></a></small>
    <div class="hidden name"><code>pandaDiffEdges.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>To determine the probability that an edge is "different" between the networks, 
we first subtracted the z-score weight values estimated by PANDA for the two networks and then determined the value of the inverse cumulative distribution for this difference. 
The product of these two probabilities represents the probability than an edge is both "supported" and "different."
We select edges for which this combined probability is greater than a threshold probability (default value is 0.8).</p>
    </div>

    <pre class="usage"><span class='fu'>pandaDiffEdges</span><span class='op'>(</span>
  <span class='va'>panda.net1</span>,
  <span class='va'>panda.net2</span>,
  threshold <span class='op'>=</span> <span class='fl'>0.8</span>,
  condition_name <span class='op'>=</span> <span class='st'>"cond.1"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>panda.net1</th>
      <td><p>vector indicating the PANDA networks of one condition or phenotype.</p></td>
    </tr>
    <tr>
      <th>panda.net2</th>
      <td><p>vector indicating the PANDA networks of another compared condition orphenotype.</p></td>
    </tr>
    <tr>
      <th>threshold</th>
      <td><p>numerical vector indicating a threshold probability to select select edges.</p></td>
    </tr>
    <tr>
      <th>condition_name</th>
      <td><p>string vector indicating the condition name of net1</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a data.frame with five columns: tf, gene, motif, Score and defined condition name(the row with "T" in this column means this egde belongs to first condition or phenotype,
"F" means this edge belongs to the second condition or phenotype)</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># refer to four input datasets files in inst/extdat</span>
<span class='va'>treated_expression_file_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"expr4_matched.txt"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>control_expression_file_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"expr10_matched.txt"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>motif_file_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"chip_matched.txt"</span>, package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>ppi_file_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"ppi_matched.txt"</span>, package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>


<span class='co'># Run PANDA for treated and control network</span>
<span class='va'>treated_all_panda_result</span> <span class='op'>&lt;-</span> <span class='fu'><a href='pandaPy.html'>pandaPy</a></span><span class='op'>(</span>expr_file <span class='op'>=</span> <span class='va'>treated_expression_file_path</span>, 
motif_file<span class='op'>=</span> <span class='va'>motif_file_path</span>, ppi_file <span class='op'>=</span> <span class='va'>ppi_file_path</span>, modeProcess<span class='op'>=</span><span class='st'>"legacy"</span>, remove_missing <span class='op'>=</span> <span class='cn'>TRUE</span> <span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Use the legacy mode to pre-process the input dataset and keep only the matched TFs or Genes</span></div><div class='output co'>#&gt; <span class='message'>Rename the content of first two columns with prefix 'reg_' and 'tar_' as there are some duplicate node names between the first two columns</span></div><div class='output co'>#&gt; <span class='message'>...Finish PANDA...</span></div><div class='input'><span class='va'>control_all_panda_result</span> <span class='op'>&lt;-</span> <span class='fu'><a href='pandaPy.html'>pandaPy</a></span><span class='op'>(</span>expr_file <span class='op'>=</span> <span class='va'>control_expression_file_path</span>, 
motif_file<span class='op'>=</span> <span class='va'>motif_file_path</span>, ppi_file<span class='op'>=</span> <span class='va'>ppi_file_path</span>, modeProcess<span class='op'>=</span><span class='st'>"legacy"</span>, remove_missing <span class='op'>=</span> <span class='cn'>TRUE</span> <span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Use the legacy mode to pre-process the input dataset and keep only the matched TFs or Genes</span></div><div class='output co'>#&gt; <span class='message'>Rename the content of first two columns with prefix 'reg_' and 'tar_' as there are some duplicate node names between the first two columns</span></div><div class='output co'>#&gt; <span class='message'>...Finish PANDA...</span></div><div class='input'>
<span class='co'># access PANDA regulatory network</span>
<span class='va'>treated_net</span> <span class='op'>&lt;-</span> <span class='va'>treated_all_panda_result</span><span class='op'>$</span><span class='va'>panda</span>
<span class='va'>control_net</span> <span class='op'>&lt;-</span> <span class='va'>control_all_panda_result</span><span class='op'>$</span><span class='va'>panda</span>

<span class='va'>merged.panda</span> <span class='op'>&lt;-</span> <span class='fu'>pandaDiffEdges</span><span class='op'>(</span><span class='va'>treated_net</span>, <span class='va'>control_net</span>, condition_name<span class='op'>=</span><span class='st'>"treated"</span><span class='op'>)</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


