<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run Python implementation PANDA in R — pandaPy • netZooR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run Python implementation PANDA in R — pandaPy" />
<meta property="og:description" content="PANDA(Passing Attributes between Networks for Data Assimilation) is a message-passing model 
to reconstruct gene regulatory network, which integrates multiple sources of biological data-including protein-protein interaction data,
gene expression data, and transcription factor binding motifs data to reconstruct genome-wide, condition-specific regulatory networks.
[(Glass et al. 2013)])
This function is designed to run the a derived PANDA implementation in Python Library &quot;netZooPy&quot; netZooPy." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/netZoo/netZooR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run Python implementation PANDA in R</h1>
    <small class="dont-index">Source: <a href='https://github.com/netZoo/netZooR/blob/master/R/PANDA.R'><code>R/PANDA.R</code></a></small>
    <div class="hidden name"><code>pandaPy.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><strong>PANDA</strong>(Passing Attributes between Networks for Data Assimilation) is a message-passing model 
to reconstruct gene regulatory network, which integrates multiple sources of biological data-including protein-protein interaction data,
gene expression data, and transcription factor binding motifs data to reconstruct genome-wide, condition-specific regulatory networks.
<a href='http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064832'>[(Glass et al. 2013)])</a>
This function is designed to run the a derived PANDA implementation in Python Library "netZooPy" <a href='https://github.com/netZoo/netZooPy'>netZooPy</a>.</p>
    </div>

    <pre class="usage"><span class='fu'>pandaPy</span><span class='op'>(</span>
  <span class='va'>expr_file</span>,
  motif_file <span class='op'>=</span> <span class='cn'>NULL</span>,
  ppi_file <span class='op'>=</span> <span class='cn'>NULL</span>,
  computing <span class='op'>=</span> <span class='st'>"cpu"</span>,
  precision <span class='op'>=</span> <span class='st'>"double"</span>,
  save_memory <span class='op'>=</span> <span class='cn'>FALSE</span>,
  save_tmp <span class='op'>=</span> <span class='cn'>TRUE</span>,
  keep_expression_matrix <span class='op'>=</span> <span class='cn'>FALSE</span>,
  modeProcess <span class='op'>=</span> <span class='st'>"union"</span>,
  remove_missing <span class='op'>=</span> <span class='cn'>FALSE</span>,
  with_header <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>expr_file</th>
      <td><p>Character string indicating the file path of expression values file, with each gene(in rows) across samples(in columns).</p></td>
    </tr>
    <tr>
      <th>motif_file</th>
      <td><p>An optional character string indicating the file path of a prior transcription factor binding motifs dataset.
When this argument is not provided, analysis will continue with Pearson correlation matrix.</p></td>
    </tr>
    <tr>
      <th>ppi_file</th>
      <td><p>An optional character string indicating the file path of protein-protein interaction edge dataset.
Also, this can be generated with a list of proteins of interest by <code><a href='sourcePPI.html'>sourcePPI</a></code>.</p></td>
    </tr>
    <tr>
      <th>computing</th>
      <td><p>'cpu' uses Central Processing Unit (CPU) to run PANDA; 'gpu' use the Graphical Processing Unit (GPU) to run PANDA. The default value is "cpu".</p></td>
    </tr>
    <tr>
      <th>precision</th>
      <td><p>'double' computes the regulatory network in double precision (15 decimal digits); 'single' computes the regulatory network in single precision (7 decimal digits) which is fastaer, requires half the memory but less accurate. The default value is 'double'.</p></td>
    </tr>
    <tr>
      <th>save_memory</th>
      <td><p>'TRUE' removes temporary results from memory. The result network is weighted adjacency matrix of size (nTFs, nGenes); 'FALSE' keeps the temporary files in memory. The result network has 4 columns in the form gene - TF - weight in motif prior - PANDA edge. PANDA indegree/outdegree of panda network, only if save_memory = FALSE. The default value is 'FALSE'.</p></td>
    </tr>
    <tr>
      <th>save_tmp</th>
      <td><p>'TRUE' saves middle data like expression matrix and normalized networks; 'FALSE' deletes the middle data. The default value is 'TURE'.</p></td>
    </tr>
    <tr>
      <th>keep_expression_matrix</th>
      <td><p>'TRUE' keeps the input expression matrix as an attribute in the result Panda object.'FALSE' deletes the expression matrix attribute in the Panda object. The default value is 'FALSE'.</p></td>
    </tr>
    <tr>
      <th>modeProcess</th>
      <td><p>'legacy' refers to the processing mode in netZooPy&lt;=0.5, 'union': takes the union of all TFs and genes across priors and fills the missing genes in the priors with zeros; 'intersection': intersects the input genes and TFs across priors and removes the missing TFs/genes. Default values is 'union'.</p></td>
    </tr>
    <tr>
      <th>remove_missing</th>
      <td><p>Only when modeProcess='legacy': remove_missing='TRUE' removes all unmatched TF and genes; remove_missing='FALSE' keeps all tf and genes. The default value is 'FALSE'.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>When save_memory=FALSE(default), this function will return a list of three items: 
         Use <code>$panda</code> to access the standard output of PANDA as data frame, which consists of four columns: 
         "TF", "Gene", "Motif" using 0 or 1 to indicate if this edge belongs to prior motif dataset, and "Score".</p>
<p>Use <code>$indegree</code> to access the indegree of PANDA network as data frame, which consists of two columns: "Gene", "Score".</p>
<p>Use <code>$outdegree</code> to access the outdegree of PANDA network as data frame, which consists of two columns: "TF", "Score".</p>         
<p>When save_memory=TRUE, this function will return a weigheted adjacency matirx of size (nTFs, nGenes), use <code>$WAMpanda</code> to access.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># take the treated TB dataset as example here.</span>
<span class='co'># refer to the datasets files path in inst/extdat</span>

<span class='va'>treated_expression_file_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"expr4_matched.txt"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>treated_expression_file_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"expr4_matched.txt"</span>,
 package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>motif_file_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"chip_matched.txt"</span>, package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>ppi_file_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"ppi_matched.txt"</span>, package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>


<span class='co'># Run PANDA for treated and control network</span>
<span class='co'># \donttest{</span>
<span class='va'>treated_all_panda_result</span> <span class='op'>&lt;-</span> <span class='fu'>pandaPy</span><span class='op'>(</span>expr_file <span class='op'>=</span> <span class='va'>treated_expression_file_path</span>, 
motif_file <span class='op'>=</span> <span class='va'>motif_file_path</span>, ppi_file <span class='op'>=</span> <span class='va'>ppi_file_path</span>, 
modeProcess<span class='op'>=</span><span class='st'>"legacy"</span>, remove_missing <span class='op'>=</span> <span class='cn'>TRUE</span> <span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Use the legacy mode to pre-process the input dataset and keep only the matched TFs or Genes</span></div><div class='output co'>#&gt; <span class='message'>Rename the content of first two columns with prefix 'reg_' and 'tar_' as there are some duplicate node names between the first two columns</span></div><div class='output co'>#&gt; <span class='message'>...Finish PANDA...</span></div><div class='input'>
<span class='co'># access PANDA regulatory network</span>
<span class='va'>treated_net</span> <span class='op'>&lt;-</span> <span class='va'>treated_all_panda_result</span><span class='op'>$</span><span class='va'>panda</span>

<span class='co'># access PANDA regulatory indegree network.</span>
<span class='va'>indegree_net</span> <span class='op'>&lt;-</span> <span class='va'>treated_all_panda_result</span><span class='op'>$</span><span class='va'>indegree</span>

<span class='co'># access PANDA regulatory outdegree networks</span>
<span class='va'>outdegree_net</span> <span class='op'>&lt;-</span> <span class='va'>treated_all_panda_result</span><span class='op'>$</span><span class='va'>outdegree</span>
<span class='co'># }</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


