<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run EGRET in R — runEgret • netZooR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run EGRET in R — runEgret" />
<meta property="og:description" content="Description:
              NOTE: Beta version. EGRET infers individual-specific gene regulatory networks using inidividual level data - 
              a genotype vcf file (v) and QBiC binding predictions (q) -  as well as population/reference level data - 
               eQTLs (b), a motif-gene prior (m), PPI network (p), and gene expression (e). An annotation file g is also used to
               map TF names to their corresponding ensemble ids." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/netZoo/netZooR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run EGRET in R</h1>
    <small class="dont-index">Source: <a href='https://github.com/netZoo/netZooR/blob/master/R/EGRET.R'><code>R/EGRET.R</code></a></small>
    <div class="hidden name"><code>runEgret.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Description:
              NOTE: Beta version. EGRET infers individual-specific gene regulatory networks using inidividual level data - 
              a genotype vcf file (v) and QBiC binding predictions (q) -  as well as population/reference level data - 
               eQTLs (b), a motif-gene prior (m), PPI network (p), and gene expression (e). An annotation file g is also used to
               map TF names to their corresponding ensemble ids.</p>
    </div>

    <pre class="usage"><span class='fu'>runEgret</span><span class='op'>(</span><span class='va'>b</span>, <span class='va'>v</span>, <span class='va'>q</span>, <span class='va'>m</span>, <span class='va'>e</span>, <span class='va'>p</span>, <span class='va'>g</span>, <span class='va'>t</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>b</th>
      <td><p>: Data frame of eQTL data, each row containing an eQTL which exist within motif regions adjacent to the eGene, with columns
TF, gene, variant position,variant chromosome, eQTL beta value.</p></td>
    </tr>
    <tr>
      <th>v</th>
      <td><p>: Data frame of VCF file containing SNPs of the individual in question</p></td>
    </tr>
    <tr>
      <th>q</th>
      <td><p>: Data frame of QBiC predictions of the effect of eQTL variants on TF binding. Each row represents an 
eQTL variant with a predicted negative (disruptive) effect on the binding of the TF corresponding to the motif in which the eQTL variant 
resides. Colums are: eQTL variant as chr[chrNum]_position, TF, adjacent eGene, QBiC binding effect size and QBiC binding effect (should be negative)</p></td>
    </tr>
    <tr>
      <th>m</th>
      <td><p>: Motif prior data frame. Each row represents an edge in the bipartite motif prior, with columns TF, gene and edge weight. 
The edge weight should be 1 or 0 based on the presence/absence of the TF motif in the promoter region of the gene.</p></td>
    </tr>
    <tr>
      <th>e</th>
      <td><p>: Gene expression data frame in which each row represents a gene and each column represents the expression of that gene in a sample.
The first column should contain gene IDs.</p></td>
    </tr>
    <tr>
      <th>p</th>
      <td><p>: PPI network data frame. Each row represents an edgem with columns TF, TF and interaction weight.</p></td>
    </tr>
    <tr>
      <th>g</th>
      <td><p>: Data frame mapping gene names to gene ids, with columns containing the gene ID the corresponding gene name.</p></td>
    </tr>
    <tr>
      <th>t</th>
      <td><p>: A string containing a name for the EGRET run. Output files will be labelled with this tag.</p>
<p>Outputs:</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>EGRET    : Predicted genotye-specific gene regulatory network saved as tag_egret.RData</p>
<p>BASELINE : A Baseline (PANDA) genotype-agnostic gene regulatory network saved as tag_panda.RData</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Inputs:</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'># Run EGRET algorithm</span>
<span class='va'>toy_qbic_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"toy_qbic.txt"</span>, package <span class='op'>=</span> <span class='st'>"netZooR"</span>, 
mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>toy_genotype_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"toy_genotype.vcf"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>toy_motif_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"toy_motif_prior.txt"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>toy_expr_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"toy_expr.txt"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>toy_ppi_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"toy_ppi_prior.txt"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>toy_eqtl_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"toy_eQTL.txt"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>toy_map_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"toy_map.txt"</span>, 
package <span class='op'>=</span> <span class='st'>"netZooR"</span>, mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>qbic</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span>file <span class='op'>=</span> <span class='va'>toy_qbic_path</span>, header <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>vcf</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='va'>toy_genotype_path</span>, header <span class='op'>=</span> <span class='cn'>FALSE</span>, sep <span class='op'>=</span> <span class='st'>"\t"</span>, 
stringsAsFactors <span class='op'>=</span> <span class='cn'>FALSE</span>, 
colClasses <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"character"</span>, <span class='st'>"numeric"</span>, <span class='st'>"character"</span>, <span class='st'>"character"</span>, <span class='st'>"character"</span>, 
<span class='st'>"character"</span>, <span class='st'>"character"</span>, <span class='st'>"character"</span>, <span class='st'>"character"</span>, <span class='st'>"character"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>motif</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='va'>toy_motif_path</span>, sep <span class='op'>=</span> <span class='st'>"\t"</span>, header <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>expr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='va'>toy_expr_path</span>, header <span class='op'>=</span> <span class='cn'>FALSE</span>, sep <span class='op'>=</span> <span class='st'>"\t"</span>, row.names <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
<span class='va'>ppi</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='va'>toy_ppi_path</span>, header <span class='op'>=</span> <span class='cn'>FALSE</span>, sep <span class='op'>=</span> <span class='st'>"\t"</span><span class='op'>)</span>
<span class='va'>qtl</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='va'>toy_eqtl_path</span>, header <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>nameGeneMap</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='va'>toy_map_path</span>, header <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>tag</span> <span class='op'>&lt;-</span> <span class='st'>"my_toy_egret_run"</span>
<span class='co'># \donttest{</span>
<span class='fu'>runEgret</span><span class='op'>(</span><span class='va'>qtl</span>,<span class='va'>vcf</span>,<span class='va'>qbic</span>,<span class='va'>motif</span>,<span class='va'>expr</span>,<span class='va'>ppi</span>,<span class='va'>nameGeneMap</span>,<span class='va'>tag</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "Initializing and validating"</div><div class='output co'>#&gt; <span class='message'>0 genes removed that were not present in motif</span></div><div class='output co'>#&gt; <span class='message'>12 motif edges removed that targeted genes missing in expression data</span></div><div class='output co'>#&gt; [1] "Verified sufficient samples"</div><div class='output co'>#&gt; <span class='message'>0 PPI edges removed that were not present in motif</span></div><div class='output co'>#&gt; [1] "Normalizing networks..."
#&gt; [1] "Learning Network..."
#&gt; [1] "Using tanimoto similarity"</div><div class='output co'>#&gt; <span class='message'>Iteration0: hamming distance =0.35159</span></div><div class='output co'>#&gt; <span class='message'>Iteration1: hamming distance =0.28205</span></div><div class='output co'>#&gt; <span class='message'>Iteration2: hamming distance =0.25188</span></div><div class='output co'>#&gt; <span class='message'>Iteration3: hamming distance =0.24416</span></div><div class='output co'>#&gt; <span class='message'>Iteration4: hamming distance =0.25299</span></div><div class='output co'>#&gt; <span class='message'>Iteration5: hamming distance =0.26142</span></div><div class='output co'>#&gt; <span class='message'>Iteration6: hamming distance =0.26807</span></div><div class='output co'>#&gt; <span class='message'>Iteration7: hamming distance =0.26952</span></div><div class='output co'>#&gt; <span class='message'>Iteration8: hamming distance =0.26802</span></div><div class='output co'>#&gt; <span class='message'>Iteration9: hamming distance =0.25899</span></div><div class='output co'>#&gt; <span class='message'>Iteration10: hamming distance =0.24379</span></div><div class='output co'>#&gt; <span class='message'>Iteration11: hamming distance =0.22374</span></div><div class='output co'>#&gt; <span class='message'>Iteration12: hamming distance =0.20081</span></div><div class='output co'>#&gt; <span class='message'>Iteration13: hamming distance =0.17714</span></div><div class='output co'>#&gt; <span class='message'>Iteration14: hamming distance =0.15418</span></div><div class='output co'>#&gt; <span class='message'>Iteration15: hamming distance =0.13283</span></div><div class='output co'>#&gt; <span class='message'>Iteration16: hamming distance =0.11353</span></div><div class='output co'>#&gt; <span class='message'>Iteration17: hamming distance =0.09643</span></div><div class='output co'>#&gt; <span class='message'>Iteration18: hamming distance =0.08151</span></div><div class='output co'>#&gt; <span class='message'>Iteration19: hamming distance =0.06862</span></div><div class='output co'>#&gt; <span class='message'>Iteration20: hamming distance =0.05758</span></div><div class='output co'>#&gt; <span class='message'>Iteration21: hamming distance =0.04819</span></div><div class='output co'>#&gt; <span class='message'>Iteration22: hamming distance =0.04023</span></div><div class='output co'>#&gt; <span class='message'>Iteration23: hamming distance =0.03352</span></div><div class='output co'>#&gt; <span class='message'>Iteration24: hamming distance =0.02789</span></div><div class='output co'>#&gt; <span class='message'>Iteration25: hamming distance =0.02316</span></div><div class='output co'>#&gt; <span class='message'>Iteration26: hamming distance =0.01921</span></div><div class='output co'>#&gt; <span class='message'>Iteration27: hamming distance =0.01591</span></div><div class='output co'>#&gt; <span class='message'>Iteration28: hamming distance =0.01317</span></div><div class='output co'>#&gt; <span class='message'>Iteration29: hamming distance =0.01089</span></div><div class='output co'>#&gt; <span class='message'>Iteration30: hamming distance =0.00899</span></div><div class='output co'>#&gt; <span class='message'>Iteration31: hamming distance =0.00742</span></div><div class='output co'>#&gt; <span class='message'>Iteration32: hamming distance =0.00612</span></div><div class='output co'>#&gt; <span class='message'>Iteration33: hamming distance =0.00504</span></div><div class='output co'>#&gt; <span class='message'>Iteration34: hamming distance =0.00415</span></div><div class='output co'>#&gt; <span class='message'>Iteration35: hamming distance =0.00342</span></div><div class='output co'>#&gt; <span class='message'>Iteration36: hamming distance =0.00281</span></div><div class='output co'>#&gt; <span class='message'>Iteration37: hamming distance =0.00231</span></div><div class='output co'>#&gt; <span class='message'>Iteration38: hamming distance =0.0019</span></div><div class='output co'>#&gt; <span class='message'>Iteration39: hamming distance =0.00156</span></div><div class='output co'>#&gt; <span class='message'>Iteration40: hamming distance =0.00128</span></div><div class='output co'>#&gt; <span class='message'>Iteration41: hamming distance =0.00105</span></div><div class='output co'>#&gt; <span class='message'>Iteration42: hamming distance =0.00087</span></div><div class='output co'>#&gt; <span class='message'>Successfully ran PANDA on 4 Genes and 6 TFs.</span>
#&gt; <span class='message'>Time elapsed:0.02seconds.</span></div><div class='output co'>#&gt; [1] "Initializing and validating"</div><div class='output co'>#&gt; <span class='message'>0 genes removed that were not present in motif</span></div><div class='output co'>#&gt; <span class='message'>12 motif edges removed that targeted genes missing in expression data</span></div><div class='output co'>#&gt; [1] "Verified sufficient samples"</div><div class='output co'>#&gt; <span class='message'>0 PPI edges removed that were not present in motif</span></div><div class='output co'>#&gt; [1] "Normalizing networks..."
#&gt; [1] "Learning Network..."
#&gt; [1] "Using tanimoto similarity"</div><div class='output co'>#&gt; <span class='message'>Iteration0: hamming distance =0.22762</span></div><div class='output co'>#&gt; <span class='message'>Iteration1: hamming distance =0.2101</span></div><div class='output co'>#&gt; <span class='message'>Iteration2: hamming distance =0.21795</span></div><div class='output co'>#&gt; <span class='message'>Iteration3: hamming distance =0.25112</span></div><div class='output co'>#&gt; <span class='message'>Iteration4: hamming distance =0.28488</span></div><div class='output co'>#&gt; <span class='message'>Iteration5: hamming distance =0.31193</span></div><div class='output co'>#&gt; <span class='message'>Iteration6: hamming distance =0.33076</span></div><div class='output co'>#&gt; <span class='message'>Iteration7: hamming distance =0.34225</span></div><div class='output co'>#&gt; <span class='message'>Iteration8: hamming distance =0.34137</span></div><div class='output co'>#&gt; <span class='message'>Iteration9: hamming distance =0.32951</span></div><div class='output co'>#&gt; <span class='message'>Iteration10: hamming distance =0.30886</span></div><div class='output co'>#&gt; <span class='message'>Iteration11: hamming distance =0.28222</span></div><div class='output co'>#&gt; <span class='message'>Iteration12: hamming distance =0.25247</span></div><div class='output co'>#&gt; <span class='message'>Iteration13: hamming distance =0.22205</span></div><div class='output co'>#&gt; <span class='message'>Iteration14: hamming distance =0.19274</span></div><div class='output co'>#&gt; <span class='message'>Iteration15: hamming distance =0.1656</span></div><div class='output co'>#&gt; <span class='message'>Iteration16: hamming distance =0.14117</span></div><div class='output co'>#&gt; <span class='message'>Iteration17: hamming distance =0.11962</span></div><div class='output co'>#&gt; <span class='message'>Iteration18: hamming distance =0.10088</span></div><div class='output co'>#&gt; <span class='message'>Iteration19: hamming distance =0.08476</span></div><div class='output co'>#&gt; <span class='message'>Iteration20: hamming distance =0.071</span></div><div class='output co'>#&gt; <span class='message'>Iteration21: hamming distance =0.05932</span></div><div class='output co'>#&gt; <span class='message'>Iteration22: hamming distance =0.04946</span></div><div class='output co'>#&gt; <span class='message'>Iteration23: hamming distance =0.04116</span></div><div class='output co'>#&gt; <span class='message'>Iteration24: hamming distance =0.0342</span></div><div class='output co'>#&gt; <span class='message'>Iteration25: hamming distance =0.02838</span></div><div class='output co'>#&gt; <span class='message'>Iteration26: hamming distance =0.02352</span></div><div class='output co'>#&gt; <span class='message'>Iteration27: hamming distance =0.01947</span></div><div class='output co'>#&gt; <span class='message'>Iteration28: hamming distance =0.0161</span></div><div class='output co'>#&gt; <span class='message'>Iteration29: hamming distance =0.0133</span></div><div class='output co'>#&gt; <span class='message'>Iteration30: hamming distance =0.01098</span></div><div class='output co'>#&gt; <span class='message'>Iteration31: hamming distance =0.00906</span></div><div class='output co'>#&gt; <span class='message'>Iteration32: hamming distance =0.00747</span></div><div class='output co'>#&gt; <span class='message'>Iteration33: hamming distance =0.00615</span></div><div class='output co'>#&gt; <span class='message'>Iteration34: hamming distance =0.00507</span></div><div class='output co'>#&gt; <span class='message'>Iteration35: hamming distance =0.00417</span></div><div class='output co'>#&gt; <span class='message'>Iteration36: hamming distance =0.00343</span></div><div class='output co'>#&gt; <span class='message'>Iteration37: hamming distance =0.00282</span></div><div class='output co'>#&gt; <span class='message'>Iteration38: hamming distance =0.00232</span></div><div class='output co'>#&gt; <span class='message'>Iteration39: hamming distance =0.0019</span></div><div class='output co'>#&gt; <span class='message'>Iteration40: hamming distance =0.00156</span></div><div class='output co'>#&gt; <span class='message'>Iteration41: hamming distance =0.00128</span></div><div class='output co'>#&gt; <span class='message'>Iteration42: hamming distance =0.00105</span></div><div class='output co'>#&gt; <span class='message'>Iteration43: hamming distance =0.00087</span></div><div class='output co'>#&gt; <span class='message'>Successfully ran PANDA on 4 Genes and 6 TFs.</span>
#&gt; <span class='message'>Time elapsed:0.02seconds.</span></div><div class='input'><span class='co'># }</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


